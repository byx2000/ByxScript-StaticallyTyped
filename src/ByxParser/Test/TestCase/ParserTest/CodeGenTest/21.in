function void f1()
{
	
}

function int f2()
{
	return 100;
}

function void f3(int n)
{
	
}

function int f4(int n)
{
	int i = n;
	return i;
}

function void f5(int a, int b, int c)
{
	
}

function int f6(int a, int b, int c)
{
	return c;
}

function void main()
{
	int a = 100;
	int b = 200;
	int c = 300;
	
	f1();
	f2();
	f3(100);
	f4(a);
	f5(100, 200, 300);
	f6(a, b, c);
	f6(100, b, 1000);
}