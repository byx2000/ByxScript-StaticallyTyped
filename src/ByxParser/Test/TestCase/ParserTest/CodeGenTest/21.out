7
3 22
0 0
0 1
1 4
2 6
3 12
3 16

0

ret

iconst 100
ret
ret

istore 0
ret

istore 0
iload 0
istore 1
iload 1
ret
ret

istore 0
istore 1
istore 2
ret

istore 0
istore 1
istore 2
iload 2
ret
ret

iconst 100
istore 0
iconst 200
istore 1
iconst 300
istore 2
call 1
call 2
pop
iconst 100
call 3
iload 0
call 4
pop
iconst 300
iconst 200
iconst 100
call 5
iload 2
iload 1
iload 0
call 6
pop
iconst 1000
iload 1
iconst 100
call 6
pop
ret